File: src/main/java/com/back/domain/game/game/dto/GameSearchResponse.java
package com.back.domain.game.game.dto;

import com.back.global.igdb.dto.IgdbGameSummaryDto;
import com.back.standard.util.TimeUt;

import com.back.global.igdb.util.IgdbImageUtil;

import java.time.LocalDate;
import java.util.List;

import java.util.Map;
import java.util.Objects;


public record GameSearchResponse(
        long igdbId,
        String name,
        String imageUrl,
        LocalDate firstReleaseDate,
        List<String> genres,
        List<String> platforms
        // developerName 추가예정
) {
//    IGDB 조회용
    public static GameSearchResponse fromDto(
            IgdbGameSummaryDto d,
            Map<Long, String> genreMap,
            Map<Long, String> platformMap
    ) {
        return new GameSearchResponse(
                d.id(),
                d.name(),
                IgdbImageUtil.cover(
                        d.cover() != null ? d.cover().imageId() : null
                ),
                TimeUt.epoch.toLocalDate(d.firstReleaseDateEpochSeconds()),
                // 장르 null 체크
                d.genres() == null ? List.of() : d.genres().stream()
                        .map(genreMap::get)
                        .filter(Objects::nonNull)
                        .toList(),
                // 플랫폼 null 체크
                d.platforms() == null ? List.of() : d.platforms().stream()
                        .map(platformMap::get)
                        .filter(Objects::nonNull)
                        .toList()
        );
    }
}

File: src/main/java/com/back/domain/game/game/dto/GameSearchResponse.kt
package com.back.domain.game.game.dto

import com.back.global.igdb.dto.IgdbGameSummaryDto
import com.back.global.igdb.util.IgdbImageUtil
import com.back.standard.util.TimeUt
import java.time.LocalDate

data class GameSearchResponse(
    val igdbId: Long,
    val name: String?,
    val imageUrl: String?,
    val firstReleaseDate: LocalDate?,
    val genres: List<String?>?,
    val platforms: List<String?>?,
) {
    companion object {
        fun fromDto(
            d: IgdbGameSummaryDto,
            genreMap: Map<Long?, String?>,
            platformMap: Map<Long?, String?>,
        ): GameSearchResponse =
            GameSearchResponse(
                igdbId = d.id,
                name = d.name,
                imageUrl = IgdbImageUtil.cover(d.cover?.imageId),
                firstReleaseDate = TimeUt.epoch.toLocalDate(d.firstReleaseDateEpochSeconds),
                genres = d.genres?.mapNotNull { genreMap[it] } ?: emptyList(),
                platforms = d.platforms?.mapNotNull { platformMap[it] } ?: emptyList(),
            )
    }
}
